### PR TITLE
Metric Exporters Prep - Rename Environment Variables

### LINKED ISSUE #420: Rename Configuration Keys Environment Variables
Following environment variables has very generic name. 

`OTEL_DOTNET_AUTO_ENABLED`
`OTEL_DOTNET_AUTO_LOAD_AT_STARTUP`
`OTEL_DOTNET_AUTO_CONSOLE_EXPORTER_ENABLED`
`OTEL_DOTNET_AUTO_INSTRUMENTATION_PLUGINS`

When auto-instrumentation is enabled for all 3 signals (traces, logs and metrics) this could cause confusion. I propose adding a `_TRACE_` to these variable names. Something like below.

`OTEL_DOTNET_AUTO_TRACE_ENABLED`
`OTEL_DOTNET_AUTO_TRACE_LOAD_AT_STARTUP`
`OTEL_DOTNET_AUTO_TRACE_CONSOLE_EXPORTER_ENABLED`
`OTEL_DOTNET_AUTO_TRACE_INSTRUMENTATION_PLUGINS`

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Fixes #420 

Renames generic environment variable to include trace in it. Main changes are in the file `ConfigurationKeys.cs`

`OTEL_DOTNET_AUTO_ENABLED`&#8594;`OTEL_DOTNET_AUTO_TRACE_ENABLED`
`OTEL_DOTNET_AUTO_LOAD_AT_STARTUP`&#8594;`OTEL_DOTNET_AUTO_LOAD_TRACER_AT_STARTUP`
`OTEL_DOTNET_AUTO_CONSOLE_EXPORTER_ENABLED`&#8594;`OTEL_DOTNET_AUTO_TRACE_CONSOLE_EXPORTER_ENABLED`
`OTEL_DOTNET_AUTO_ENABLED_INSTRUMENTATIONS`&#8594;`OTEL_DOTNET_AUTO_TRACE_ENABLED_INSTRUMENTATIONS`
`OTEL_DOTNET_AUTO_DISABLED_INSTRUMENTATIONS`&#8594;`OTEL_DOTNET_AUTO_TRACE_INSTRUMENTATION_PLUGINS`
`OTEL_DOTNET_AUTO_INSTRUMENTATION_PLUGINS`&#8594;`OTEL_DOTNET_AUTO_TRACE_INSTRUMENTATION_PLUGINS`
`OTEL_DOTNET_AUTO_ADDITIONAL_SOURCES`&#8594;`OTEL_DOTNET_AUTO_TRACE_ADDITIONAL_SOURCES`

### CODE REVIEW

**Review by pellared — APPROVED** [review]
Can you update the `CHANGELOG.md` and list the changes? 

If we merge it, the next release should be `0.2.0-beta.1` (I have no problems with it) 😄

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,15 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - Adds MongoDB instrumentation support from .NET Core 3.1+.
-
+- Raname generic environment variables to include trace.
```
Comment (Kielek) [inline CHANGELOG.md:13]: I think that we need `Changed` header

Comment (rajkumar-rangaraj) [reply CHANGELOG.md:16]: Fixed e6068984182957fc5e6fdd49f88e2735ba249c28

**Review by Kielek — APPROVED** [review]

**Comment by nrcventura** [comment]
Some questions that I think we should answer:

1. Do we want to have telemetry specific settings for most of the things?
2. Do we think that we should keep things simpler for now and just have settings that apply to all of the telemetry types?
3. Do we need both types of settings to handle the different use cases?

I think that the answers to these questions will vary based on the different user scenarios we can envision. 

I think that the approach taken in this PR makes the most sense for now, because it is much easier to add a more global setting later than it is to split that setting into multiple telemetry-specific settings later.

**Comment by pellared** [comment]
> I think that the approach taken in this PR makes the most sense for now, because it is much easier to add a more global setting later than it is to split that setting into multiple telemetry-specific settings later.

I agree. Also, I find more fine-grained settings often more useful.

**Review by nrcventura — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Pull request merged**